File: Week 1 - C/mario-less/mario.c
// This program create a right-aligned "#" pyramid
#include <cs50.h>
#include <stdio.h>

// Here are the function prototypes
void print_space(int height, int row);
void print_hash(int i);

int main(void)
{
    int height = 0;
    // To make sure  the user's input is valid
    while (height < 1)
    {
        height = get_int("Height: ");
    }
    // Prompt the user for the height
    for (int i = 1; i <= height; i++)
    {
        print_space(height, i);
        print_hash(i);
        printf("\n");
    }
}

// Funtion to print the spaces
void print_space(int height, int row)
{
    for (int j = 0; j < height - row; j++)
    {
        printf(" ");
    }
}

// Function to print the hash (#)
void print_hash(int i)
{
    for (int k = 0; k < i; k++)
    {
        printf("#");
    }
}

File: Week 6 - Python/sentimental-mario-less/mario.py
def print_space(height, row):
    """Print spaces for the given row"""
    print(' ' * (height - row), end='')

def print_hash(count):
    """Print hashes for the given count"""
    print('#' * count, end='')

def main():
    height = 0
    # To make sure the user's input is valid
    while height < 1 or height > 8:
        try:
            height = int(input("Height: "))
            if height >= 1 and height <= 8:
                break
        except ValueError:
            continue

    # Prompt the user for the height and build the pyramid
    for i in range(1, height + 1):
        print_space(height, i)
        print_hash(i)
        print()  # Move to the next line

main()

